astropy/utils/parsing.py

```diff
@@ -0,0 +1,153 @@
+# -*- coding: utf-8 -*-
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+Wrappers for PLY to provide thread safety.
+"""
+
+import contextlib
+import functools
+import re
+import os
+import threading
+
+
+__all__ = ['lex', 'TAB_HEADER', 'ThreadSafeParser', 'yacc']
```

Comment:
I think @mhvk wants TAB_HEADER removed as per comment above?
```suggestion
__all__ = ['lex', 'ThreadSafeParser', 'yacc']
```